api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,22 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion is the deletion state of the Machine.
+	// +optional
+	Deletion MachineStatusDeletion `json:"deletion,omitempty"`
 }
 
 // ANCHOR_END: MachineStatus
 
+// MachineStatusDeletion is the deletion state of the Machine.
+type MachineStatusDeletion struct {
+	// NodeDrainStartTime is the time when the drain of the node started.
+	NodeDrainStartTime *metav1.Time `json:"nodeDrainStartTime,omitempty"`
```

Comment:
```suggestion
        +optional
	NodeDrainStartTime *metav1.Time `json:"nodeDrainStartTime,omitempty"`
```
(same below)
NOTE: it is not necessary to make the field not required, but as far as I remember we agreed to add it to be really explicit about our intention, cc @sbueringer

Reply:
Yeah we should always add optional or required